homeassistant/components/fan/__init__.py

```diff
@@ -91,6 +94,12 @@ async def async_setup(hass, config: dict):
         "async_set_direction",
         [SUPPORT_DIRECTION],
     )
+    component.async_register_entity_service(
+        SERVICE_SET_SPEED_PERCENTAGE,
+        {vol.Required(ATTR_SPEED): vol.Number()},
+        "async_set_speed_percentage",
+        [SUPPORT_SET_SPEED],
```

Comment:
```suggestion
        [SERVICE_SET_SPEED_PERCENTAGE],
```

Reply:
That's what i did originally, but because of the mapping functions we want both this and `async_set_speed` to always be available. I tried SERVICE_SET_SPEED | SERVICE_SET_SPEEED_PERCETAGE which obviously didn't work. Is what i wanted `[SERVICE_SET_SPEED` SERVICE_SET_SPEEED_PERCETAGE ]` maybe?

Reply:
From top of my head, yes.